src/extractor/scripting_environment_lua.cpp

```diff
@@ -200,10 +208,8 @@ void LuaScriptingEnvironment::InitContext(LuaScriptingContext &context)
          luabind::class_<osmium::WayNodeList>("WayNodeList").def(luabind::constructor<>()),
          luabind::class_<osmium::NodeRef>("NodeRef")
              .def(luabind::constructor<>())
-             // Dear ambitious reader: registering .location() as in:
-             // .def("location", +[](const osmium::NodeRef& nref){ return nref.location(); })
-             // will crash at runtime, since we're not (yet?) using libosnmium's
-             // NodeLocationsForWays cache
+             .def("location", +[](osmium::NodeRef &nref){
+                 return extraction_containers->getLocationForNode(nref);})
```

Comment:
Instead of a static var, you can make a ScriptingEnvironment-local function (see e.g. lonToDouble at the top) and register it via `&my_function` with Luabind.